sktime/decomposition/koopman.py

```diff
@@ -0,0 +1,575 @@
+            :keyprefix: vampscore-
+        """
+        if test_model is None:
+            test_model = self
+        Uk = self.singular_vectors_left[:, 0:self.output_dimension]
+        Vk = self.singular_vectors_right[:, 0:self.output_dimension]
+        res = None
+        if score_method == 'VAMP1' or score_method == 'VAMP2':
+            A = spd_inv_sqrt(Uk.T.dot(test_model.cov_00).dot(Uk), epsilon=self.epsilon)
+            B = Uk.T.dot(test_model.cov_0t).dot(Vk)
+            C = spd_inv_sqrt(Vk.T.dot(test_model.cov_tt).dot(Vk), epsilon=self.epsilon)
+            ABC = mdot(A, B, C)
+            if score_method == 'VAMP1':
+                res = np.linalg.norm(ABC, ord='nuc')
+            elif score_method == 'VAMP2':
+                res = np.linalg.norm(ABC, ord='fro')**2
+        elif score_method == 'VAMPE':
+            Sk = np.diag(self.singular_values[0:self.output_dimension])
+            res = np.trace(2.0 * mdot(Vk, Sk, Uk.T, test_model.cov_0t)
+                           - mdot(Vk, Sk, Uk.T, test_model.cov_00, Uk, Sk, Vk.T, test_model.cov_tt))
+        else:
+            raise ValueError('"score" should be one of VAMP1, VAMP2 or VAMPE')
+        assert res is not None
+        # add the contribution (+1) of the constant singular functions to the result
+        return res + 1
+
+    def expectation(self, observables, statistics, lag_multiple=1, observables_mean_free=False,
+                    statistics_mean_free=False):
+        r"""Compute future expectation of observable or covariance using the approximated Koopman operator.
+
+        Parameters
+        ----------
+        observables : np.ndarray((input_dimension, n_observables))
+            Coefficients that express one or multiple observables in
+            the basis of the input features.
+
+        statistics : np.ndarray((input_dimension, n_statistics)), optional
+            Coefficients that express one or multiple statistics in
+            the basis of the input features.
+            This parameter can be None. In that case, this method
+            returns the future expectation value of the observable(s).
+
+        lag_multiple : int
+            If > 1, extrapolate to a multiple of the estimator's lag
+            time by assuming Markovianity of the approximated Koopman
+            operator.
+
+        observables_mean_free : bool, default=False
+            If true, coefficients in `observables` refer to the input
+            features with feature means removed.
+            If false, coefficients in `observables` refer to the
+            unmodified input features.
+
+        statistics_mean_free : bool, default=False
+            If true, coefficients in `statistics` refer to the input
+            features with feature means removed.
+            If false, coefficients in `statistics` refer to the
+            unmodified input features.
+
+        Notes
+        -----
+        A "future expectation" of a observable g is the average of g computed
```

Comment:
should this be connected to the forward/backward paradigm

Reply:
definitely connected but also only valid in the vamp/tica case id imagine